server/auth/sso/sso.go

```diff
@@ -263,8 +263,7 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if s.customClaimName != "" {
 		groups, err = c.GetCustomGroup(s.customClaimName)
 		if err != nil {
-			w.WriteHeader(401)
-			return
+			log.Warn(err)
```

Comment:
This error handling seems inconsistent with other similar places above

Reply:
Hey terry, thank you for taking a look at this PR! 
I gave a more descriptive explanation in the linked issue, so if possible take a look there as well :) 
In general, the idea here is that by throwing a 401 exception when failing to retrieve a `customGroup` we are being inconsistent with the default behaviour (argo workflows doesn't throw 401 Unauthorized if the groups claim is missing)

This is causing a bunch of issues with my integration with Azure, which uses "roles" as a way to authorise users to an App Registration. If the roles claim is empty it shouldn't throw 401 Unauthorized, but evaluate it as empty (as it's currently the case for groups)

Reply:
cc @liafizan @alexec to take a look as well

Reply:
Hey @terrytangyuan , any update on this? As we're about deploying argo workflows to production we'd really need this feature to work properly in order to grant granular access to users

Reply:
I see that there was a problem of inconsistency and this is addressing it. I'm gathering that in the case where customGroupClaimName is not defined, c.Groups may in fact be set to empty and nothing really happens if that's the case. So, I think this seems okay to me.